pkg/kubelet/kubelet.go

```diff
@@ -1853,6 +1864,12 @@ func (kl *Kubelet) syncTerminatingPod(ctx context.Context, pod *v1.Pod, podStatu
 		return err
 	}
 
+	// Once the containers are stopped, we can stop probing for liveness and readiness.
+	// TODO: once a pod is terminal, certain probes (liveness exec) could be stopped immediately after
+	//   the detection of a container shutdown or (for readiness) after the first failure. Tracked as
+	//   https://github.com/kubernetes/kubernetes/issues/107894 although may not be worth optimizing.
+	kl.probeManager.RemovePod(pod)
```

Comment:
why is this done before the container status is checked a few lines down and runningContainers identified?

Reply:
killPod (a few lines up) is guaranteed to block. Once killPod returns then all the containers are terminated.